File: Base/Workspace/CSharp/Domain/Base/Workspace/Data/PushResponse.cs
namespace Allors.Workspace.Data
{
    public class PushResponse : ErrorResponse
    {
        public bool hasErrors { get; set; }

        public string errorMessage { get; set; }

        public string[] versionErrors { get; set; }

        public string[] accessErrors { get; set; }

        public string[] missingErrors { get; set; }

        public PullResponseDerivationError[] derivationErrors { get; set; }

        public PushResponseNewObject[] newObjects { get; set; }
    }
}

File: Domains/Base/Workspace/Typescript/Angular/src/allors/domain/base/data/responses/PushResponse.ts
import { ErrorResponse } from './ErrorResponse';
import { PullResponseDerivationError } from './PullResponseDerivationError';
import { PushResponseNewObject } from './PushResponseNewObject';

export interface PushResponse extends ErrorResponse {
    hasErrors: boolean;
    errorMessage?: string;
    versionErrors?: string[];
    accessErrors?: string[];
    missingErrors?: string[];
    derivationErrors?: PullResponseDerivationError[];

    newObjects?: PushResponseNewObject[];
}
